dot/core/messages.go

```diff
@@ -5,16 +5,53 @@ package core
 
 import (
 	"errors"
-
-	"github.com/libp2p/go-libp2p-core/peer"
+	"fmt"
 
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/peerset"
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/lib/runtime"
 	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/libp2p/go-libp2p-core/peer"
 )
 
+func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt runtime.Instance,
+	tx types.Extrinsic) (validity *transaction.Validity, valid bool, err error) {
+	s.storageState.Lock()
+	defer s.storageState.Unlock()
+
+	ts, err := s.storageState.TrieState(&head.StateRoot)
```

Comment:
nit to unlock it earlier?

```suggestion
	s.storageState.Lock()
	ts, err := s.storageState.TrieState(&head.StateRoot)
	s.storageState.Unlock()
```

Reply:
I think this change should be fine